src/components/App.jsx

```diff
@@ -69,9 +72,10 @@ class ContactsApp extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.state.displayedContact) {
+    if (this.state.displayedContact && !this.state.isDisplayedContactUpdated) {
```

Comment:
I'm not quite sure that using a flag here would work :|. If the contact is updated more than once, I think we'll only see the first update reflected, no?

Can I suggest we do something that's pretty close to what the [react doc](https://reactjs.org/docs/react-component.html#componentdidupdate) suggests, ie. 

```
const previouslyDisplayedContact = prevProps.contacts.find(contact => contact._id === state.displayedContact._id)
const displayedContact = this.contacts.find(contact => contact._id === state.displayedContact._id)
if (displayedContact._rev !== previouslyDisplayedContact._rev) { /* call setState here */ }
```

Probably not exactly like this, but you get the general idea.